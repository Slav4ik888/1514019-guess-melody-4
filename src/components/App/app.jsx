import React from 'react';
import PropTypes from 'prop-types';
import WelcomeScreen from '../WelcomeScreen/welcome-screen.jsx';

const welcomeButtonHandler = () => console.log(`Кликнули на большую кнопку`);

const App = (props) => {

  const {errorsValue} = props;

  return (
    <WelcomeScreen
      errorsValue={errorsValue}
      onWelcomeButtonClick={welcomeButtonHandler}
    />
  );
};


App.propTypes = {
  errorsValue: PropTypes.number.isRequired
};

export default App;
