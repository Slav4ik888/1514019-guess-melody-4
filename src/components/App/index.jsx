import React, { Component } from 'react';
import WelcomeScreen from '../WelcomeScreen/index.jsx';


class App extends React.PureComponent {

    render() {
        const { errorValue } = this.props;
        return (
            <WelcomeScreen errorValue={errorValue} />
        )
    }

}

export default App;
