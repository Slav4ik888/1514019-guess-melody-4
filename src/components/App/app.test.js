import React from 'react';
import renderer from 'react-test-renderer';
import App from './app';

const errorsValue = [1, 4, -7, 127];

describe(`index рендерит <App /> `, () => {
  it(`index рендерит <App /> `, () => {
    const tree = renderer
      .create(<App errorsValue={errorsValue} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`index рендерит <App /> errorsValue = null`, () => {
    const tree = renderer
      .create(<App errorsValue={null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
