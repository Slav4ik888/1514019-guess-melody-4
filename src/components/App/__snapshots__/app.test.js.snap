// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> рендерит WelcomeScreen <App /> рендерит WelcomeScreen 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      1
      4
      -7
      127
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;

exports[`<App /> рендерит WelcomeScreen <App /> рендерит errorsValue = null 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;

exports[`index рендерит <App />  index рендерит <App />  1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
      1
      4
      -7
      127
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;

exports[`index рендерит <App />  index рендерит <App /> errorsValue = null 1`] = `
<section
  className="welcome"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <button
    className="welcome__button"
    onClick={[Function]}
  >
    <span
      className="visually-hidden"
    >
      Начать игру
    </span>
  </button>
  <h2
    className="welcome__rules-title"
  >
    Правила игры
  </h2>
  <p
    className="welcome__text"
  >
    Правила просты:
  </p>
  <ul
    className="welcome__rules-list"
  >
    <li>
      Нужно ответить на все вопросы.
    </li>
    <li>
      Можно допустить 
       ошибки.
    </li>
  </ul>
  <p
    className="welcome__text"
  >
    Удачи!
  </p>
</section>
`;
